### PR TITLE
[Rest API] Add task state; Add job's retry details; Refine job config

### PR DESCRIPTION
**Pull request by sunqinzheng**
- add task state field
- add job's retry details field
- refine job config api
    - use request's Accept header to determine whether json or yaml will be sent.
    - BREAK CHANGE: fix the bug that old job config api may return an escaped json string instead of a json object

### CODE REVIEW

File: docs/rest-server/API.md [hunk 1]
```diff
@@ -440,10 +440,16 @@ Status: 200
     createdTime: "createdTimestamp",
     completedTime: "completedTimestamp",
     executionType: "executionType",
-    // Sum of succeededRetriedCount, transientNormalRetriedCount,
-    // transientConflictRetriedCount, nonTransientRetriedCount,
-    // and unKnownRetriedCount
-    retries: retriedCount,
+    // sum of retries
+    retries: retries,
+    retryDetails: {
```
Comment (yqwang-ms) [inline docs/rest-server/API.md]: better name to be retryBreakdown?

File: docs/rest-server/API.md [hunk 2]
```diff
@@ -440,10 +440,16 @@ Status: 200
...
+      // Failed caused by user's code
+      user: userRetries,
+      // System failed, and it can ensure that it will success within a finite retry times
+      system: systemRetries,
+      // System cannot get required resource to run.
```
Comment (yqwang-ms) [inline docs/rest-server/API.md]: "Job cannot get required resource to run within timeout"

File: docs/rest-server/API.md [hunk 3]
```diff
@@ -440,10 +440,16 @@ Status: 200
...
-    // Sum of succeededRetriedCount, transientNormalRetriedCount,
-    // transientConflictRetriedCount, nonTransientRetriedCount,
-    // and unKnownRetriedCount
-    retries: retriedCount,
+    // sum of retries
+    retries: retries,
+    retryDetails: {
+      // Failed caused by user's code
+      user: userRetries,
+      // System failed, and it can ensure that it will success within a finite retry times
```
Comment (yqwang-ms) [inline docs/rest-server/API.md]: "Job failed due to system/plaform error"

File: docs/rest-server/API.md [hunk 4]
```diff
@@ -440,10 +440,16 @@ Status: 200
...
     completedTime: "completedTimestamp",
     executionType: "executionType",
-    // Sum of succeededRetriedCount, transientNormalRetriedCount,
-    // transientConflictRetriedCount, nonTransientRetriedCount,
-    // and unKnownRetriedCount
-    retries: retriedCount,
+    // sum of retries
+    retries: retries,
+    retryDetails: {
+      // Failed caused by user's code
```
Comment (yqwang-ms) [inline docs/rest-server/API.md]: "Job failed due to user error"

File: docs/rest-server/API.md (hunk 2 above)
Comment (sunqinzheng) [reply docs/rest-server/API.md]: fixed

File: docs/rest-server/API.md (hunk 3 above)
Comment (sunqinzheng) [reply docs/rest-server/API.md]: fixed

File: docs/rest-server/API.md (hunk 4 above)
Comment (sunqinzheng) [reply docs/rest-server/API.md]: fixed

File: docs/rest-server/API.md (hunk 1 above)
Comment (sunqinzheng) [reply docs/rest-server/API.md]: I think "details" is better...

File: docs/rest-server/API.md [hunk 5]
```diff
@@ -440,10 +440,16 @@ Status: 200
...
     completedTime: "completedTimestamp",
     executionType: "executionType",
-    // Sum of succeededRetriedCount, transientNormalRetriedCount,
-    // transientConflictRetriedCount, nonTransientRetriedCount,
-    // and unKnownRetriedCount
-    retries: retriedCount,
+    // sum of retries
+    retries: retries,
+    retryDetails: {
+      // Job cannot get required resource to run within timeout
```
Comment (yqwang-ms) [inline docs/rest-server/API.md]: wrong comment location?

Comment (sunqinzheng) [reply docs/rest-server/API.md]: fixed

File: docs/rest-server/API.md [hunk 6]
```diff
@@ -440,10 +440,16 @@ Status: 200
...
-    // transientConflictRetriedCount, nonTransientRetriedCount,
-    // and unKnownRetriedCount
-    retries: retriedCount,
+    // sum of retries
+    retries: retries,
+    retryDetails: {
+      // Job failed due to user or unknown error
+      user: userRetries,
+      // Job failed due to system/plaform error
+      system: systemRetries,
```
Comment (yqwang-ms) [inline docs/rest-server/API.md]: "platform" is better than "system" due to:
1. More accurate, user job may also be a "system"
2. Our PAI is "platform" for AI, not "system" for AI
3. There are PaaS, but no "system" as a Service

so can you replace all "system" words with "platform" in this PR?

Comment (sunqinzheng) [reply docs/rest-server/API.md]: I find that "system" is used in rest api document (https://github.com/Microsoft/pai/blob/master/docs/rest-server/API.md#post-token) multiple times. Shall we change them as well?

Comment (yqwang-ms) [reply docs/rest-server/API.md]: Seems there is no need to distingush with user in that doc, so no ambiguity in the doc (the "system" in the doc implies PAI).
So, it is better to replace "in the system" to "in PAI", but it is not necessary, it depends on you :)

Comment (sunqinzheng) [reply docs/rest-server/API.md]: fixed

**Comment by sunqinzheng** [comment]
#2382

**Pull request merged**